components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/model/UserStoreAttributeMappings.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.UserStoreOperation;
+
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Data object to hold attribute mappings for all user store types.
+ */
+public class UserStoreAttributeMappings {
+
+    private Map<String, UserStoreAttributeDO> defaultUserStoreAttrMapping;
+    private static final Log LOG = LogFactory.getLog(UserStoreAttributeMappings.class);
+    private final Map<String, Map<String, UserStoreAttributeDO>> userStoreAttrMappings = new HashMap<>();
+
+    /**
+     * To get default user store attribute mappings.
+     *
+     * @return Map of attribute mappings.
+     */
+    public Map<String, UserStoreAttributeDO> getDefaultUserStoreAttributeMapping() {
+
+        return Collections.unmodifiableMap(defaultUserStoreAttrMapping);
+    }
+
+    /**
+     * Set default attribute mappings.
+     *
+     * @param defaultUserStoreAttrMapping Map of default attribute mappings.
+     */
+    public void setDefaultUserStoreAttributeMapping(Map<String, UserStoreAttributeDO> defaultUserStoreAttrMapping) {
+
+        this.defaultUserStoreAttrMapping = defaultUserStoreAttrMapping;
+    }
+
+    /**
+     * Get attribute mappings changes for the given user store type.
+     *
+     * @param userStoretype User store type.
+     * @return Map of attribute uri and attribute mappings of the given user store type.
+     */
+    public Map<String, UserStoreAttributeDO> getUserStoreAttributeMappings(String userStoretype) {
+
+        if (userStoreAttrMappings.containsKey(userStoretype)) {
+            return Collections.unmodifiableMap(userStoreAttrMappings.get(userStoretype));
+        }
+        /*
+         * If details are not available for the given user store type, return default values.
+         * Add a new userstore attribute mappings file to conf/attributes/userstore directory to
+         * configure correct attribute mappings for the userstore.
+         */
+        LOG.warn(String.format("No record found for the given userstore: %s. Default attribute mappings are " +
+                        "returning. Please add userstore attribute mappings file for the userstore %s " +
+                        "to conf/attributes/userstore directory to configure correct attribute mappings.",
+                userStoretype, userStoretype));
+        return defaultUserStoreAttrMapping;
+    }
+
+    /**
+     * Set all available user store attribute mappings.
+     *
+     * @param availableUserStoreAttrMappings Map of user store type and their attribute mappings changes.
+     */
+    public void setUserStoreAttributeMappings(Map<String, Map<String, ChangedUserStoreAttributeDO>>
+                                                      availableUserStoreAttrMappings) {
+
+        for (String userStoreType : availableUserStoreAttrMappings.keySet()) {
```

Comment:
This method accesses the value of a Map entry, using a key that was retrieved from a keySet iterator. It is more efficient to use an iterator on the entrySet of the map, to avoid the Map.get(key) lookup.

Reply:
fixed in https://github.com/wso2/carbon-identity-framework/pull/3715